### PR TITLE
Elasticsearch: remove already-disabled code

### LINKED ISSUE #48741: ElasticSearch: Cleanup of code for not-supported versions
Continuation of https://github.com/grafana/grafana/issues/48496. After we removed the support for versions after their EOL, we need to clean up the code and adjust testing.

### PR DESCRIPTION
**Pull request by gabor**
Fixes https://github.com/grafana/grafana/issues/48741

in Grafana9.0.0 we removed support for elasticsearch versions after their end-of-life (see https://github.com/grafana/grafana/pull/48715 ). 

this allows us to cleanup&simplifiy some code, which happens in this PR.

NOTE: i focused on just doing the cleanup/code-removal and nothing else. one thing we should do afterwards is: currently when you create a new elastic datasource, it defaults to `5.0.0`. it should default to `8.0.0`)

how to test:
- there are no specific features we should test with, in general use the elastic-search datasource a little and see if anything breaks.
- try it with both elastic7 and elastic8, this is how:
    - `make devenv sources=elastic` .. this runs with elastic-8.2.0
    - `make devenv sources=elastic elastic_version=7.12.1` ... this runs with elastic-7.12.1

### CODE REVIEW

File: public/app/plugins/datasource/elasticsearch/configuration/ElasticDetails.tsx [hunk 1]
```diff
@@ -82,8 +82,7 @@ export const ElasticDetails = ({ value, onChange }: Props) => {
             options={[customOption, ...esVersions].filter(isTruthy)}
             onChange={(option) => {
               const maxConcurrentShardRequests = getMaxConcurrenShardRequestOrDefault(
-                value.jsonData.maxConcurrentShardRequests,
-                option.value!
+                value.jsonData.maxConcurrentShardRequests
```
[the remark this answers is not in the document]
Comment (ivanahuckova) [reply public/app/plugins/datasource/elasticsearch/configuration/ElasticDetails.tsx:85]: And also on line 101 `{gte(value.jsonData.esVersion, '5.6.0') && (`, we can probably remove this check as all supported version are higher than 5.6.0, right?

[the remark this answers is not in the document]
Comment (ivanahuckova) [reply public/app/plugins/datasource/elasticsearch/configuration/ElasticDetails.tsx:85]: The same is on line 141 `gte(value.jsonData.esVersion, '6.6.0')`

**Review by ivanahuckova — COMMENTED** [review]
Left couple comments below. 

Also, not sure if this is strictly related to this PR or if we want to tackle it in different PR, but we should update version to supported version in tests and documentations (found bunch of pre-7.10 esVersions there.

File: public/app/plugins/datasource/elasticsearch/configuration/ElasticDetails.tsx (hunk 1 above)
[the remark this answers is not in the document]
Comment (gabor) [reply public/app/plugins/datasource/elasticsearch/configuration/ElasticDetails.tsx:85]: you're correct here, but if possible i'd do it in a separate PR. i'm trying to keep the size small here, as long as it's possible.

[the remark this answers is not in the document]
Comment (gabor) [reply public/app/plugins/datasource/elasticsearch/configuration/ElasticDetails.tsx:85]: about the `gte` calls.. good catch! removed those checks.

[the remark this answers is not in the document]
Comment (gabor) [reply public/app/plugins/datasource/elasticsearch/configuration/ElasticDetails.tsx:85]: > Also, if `maxConcurrentShardRequests` doesn't depend on version, do we need to set it here?

i'm not sure, would delay it to a separate PR 😄 

if you check the body of that function, it's a strange thing, it does an `if it's 256 let it be 5` thing...

[the remark this answers is not in the document]
Comment (svennergr) [reply public/app/plugins/datasource/elasticsearch/configuration/ElasticDetails.tsx:85]: > if you check the body of that function, it's a strange thing, it does an if it's 256 let it be 5 thing... 

Why are we doing this `256 will be 5` thing? Isn't this also some old/compat. thing?

**Review by svennergr — APPROVED** [review]
LGTM

File: public/app/plugins/datasource/elasticsearch/configuration/ElasticDetails.tsx (hunk 1 above)
[the remark this answers is not in the document]
Comment (gabor) [reply public/app/plugins/datasource/elasticsearch/configuration/ElasticDetails.tsx:85]: yes, probably :) i just don't want to do any functional changes, so i'd say we address that in a separate PR.

**Comment by heyrod** [comment]
Hi, I can't figure out why that was required. We are now planning the Grafana upgrade to the latest 9.x version but most of our data sources are built on Elastic 7.5., whci we cannot upgrade for some reasons.
In our test env we see that Elastic data sources just stop working (we have hundreds of them), defaulting to 7.0.0 version in data source settings after the upgrade. Selecting 7.10.x brings them back to life but we are not sure that everything will work as before.

**Pull request merged**